src/control/lib/control/request.go

```diff
@@ -97,14 +148,58 @@ func (r *msRequest) isMSRequest() bool {
 	return true
 }
 
-// timeoutRequest is an embeddable struct to implement the
-// timeoutGetter interface. Embedding requests can use it
-// to set an optional per-request timeout.
-type timeoutRequest struct {
-	RequestTimeout time.Duration
+// retryableRequest is the default implementation of the retryer interface.
+type retryableRequest struct {
+	// retryInterval is an optional interval to override the default.
+	retryInterval time.Duration
+	// retryMaxTries is an optional max number of retry attempts.
+	retryMaxTries uint
+	// retryTestFn defines a function that can test an error for retryability;
+	// if not set, the assumption is that the request is retryable because it
+	// embeds this type.
+	retryTestFn func(error, uint) bool
+	// retryFun defines a function that will run on every retry iteration.
+	retryFn func(context.Context, uint) error
+}
+
+// retryAfter returns a channel to be waited on in order to pause
+// a goroutine's execution for some duration. For safety, the
+// channel should be waited on in a select with a context in order to
+// avoid blocking a goroutine forever.
```

Comment:
Another comment that seems a bit more general, about how callers should behave with the interface.

Reply:
fixed